jujugui/static/gui/src/app/app.js

```diff
@@ -1614,6 +1620,15 @@ YUI.add('juju-gui', function(Y) {
      *
      */
     checkUserCredentials: function(req, res, next) {
```

Comment:
This means that every time the route changes (which is every time state changes) this method is called. The methods in the `gisf` block below are not idempotent but the values which they are setting should not change between routes. There is some micro performance improvements to be had here if one wanted to set up some sort of flag.